src/praisonai/tests/unit/test_sequential_tool_calling.py

```
@@ -0,0 +1,616 @@
+        )
+        
+        result = agent.chat("Get Microsoft stock and multiply it by 5")
+        
+        assert "500" in result
+        assert mock_completion.call_count == 3
+    
+    @patch('litellm.completion')
+    def test_sequential_with_streaming(self, mock_completion):
+        """Test sequential tool calling with streaming enabled."""
+        # For streaming, we need different mock structure
+        def streaming_side_effect(*args, **kwargs):
```

Ruff (0.14.14)

[warning] Unused function argument: `args`

(ARG001)